streamer/src/nonblocking/quic.rs

```diff
@@ -65,10 +65,10 @@ use {
         task::JoinHandle,
         time::{sleep, timeout},
     },
+    tokio_util::sync::CancellationToken,
 };
 
-const WAIT_FOR_STREAM_TIMEOUT: Duration = Duration::from_millis(100);
-pub const DEFAULT_WAIT_FOR_CHUNK_TIMEOUT: Duration = Duration::from_secs(10);
+pub const DEFAULT_WAIT_FOR_CHUNK_TIMEOUT: Duration = Duration::from_secs(1);
```

Comment:
old timeout makes no sense - a validator has 1.6s worth of leader slots, what are we waiting for 10 seconds?

Reply:
> Note 1 < 1.6s.

Yes, this is intentional. Waiting _ten seconds_ for a single _chunk_ of _one_ transaction makes no sense. The only way it can happen is if the peer is being intentionally malicious and has opened one (or more) streams and is not sending data in them. 

>  it does forward if passing the leader slot

I'll check this but hopefully we don't do it for that long because that would also make no sense

Reply:
> I would like to keep the old timeout to reduce the net change impact. If we are changing it, maybe a separate PR?

To do this I'd have to make it do the old thing of sleeping in WAIT_FOR_CHUNK_TIMEOUT / 10 intervals or the tests take forever. So I'd have to add back the old code complexity and create * 10 the number of timeout futures - in a PR trying to reduce the number of timeout futures. To me "this is what we used to do" is not a good enough argument to justify the complexity and worse perf.

Reply:
what if you reduce the timeout in a predecessor pr?

Reply:
or just separate commit and we merge w/o squashing

Reply:
wait, is not squashing even an option? 🤯

Reply:
i know a guy who can make it one, temporarily

Reply:
done, split in two commits, do your magic now trent

Reply:
I am questioning the 1s is too small and a lot of more txns may timeout due to this. At least it need to be within the leader slot. We do need to be careful fixing one issue and introducing other potential issues.

Reply:
Could you walk me through what would timeout and for what reason?

Reply:
<img width="1412" alt="Screenshot 2024-10-15 at 10 52 34 AM" src="https://github.com/user-attachments/assets/bfa9d476-0a24-4769-8761-b316ce763860">
see this in the mainnet with the current 10s timeout value. The stream can timeout due to network conditions (client, in between, server).  I also wonder why the cancel mechanism you implemented here cannot be used.

Reply:
> see this in the mainnet with the current 10s timeout value. The stream can timeout due to network conditions (client, in between, server).

We want to timeout garbage connections as soon as possible, we're trying to build a chain that goes fast. If you're sending txs from your dialup at home you should get disconnected.

Also this is a timeout for _one single stream_, a connection can still send other streams. So in case of legitimate packet loss (one or a few transactions), we wait up to 1s, otherwise we move on and keep reading other transactions from the same peer, who will receive a reset stream error and presumably retry the failed transactions.

What you're seeing in those metrics is probably peers timing out because the whole connection times out, and then because we're currently extremely bad and read many streams in parallel, we increment the metric many times.

So tldr: on legit packet loss, 1s seems more than enough (assuming decent RTT <=200ms). When a peer falls off the planet and the connection just stops sending any data, timing out as soon as possible is the best thing we can do to allocate resources to peers who are actually sending transactions.

> I also wonder why the cancel mechanism you implemented here cannot be used. 

I will let you role play as hacker for one day to answer that question!

Reply:
I am not comfortable with the big change of the timeout value. Can we set it to 2s for now as an interim approach and observe in the production of its impacts?